test/test_logging_output_format.py

```diff
@@ -35,55 +38,59 @@ def test_logging_output_format():
     env_long['RCUTILS_CONSOLE_OUTPUT_FORMAT'] = \
         '[{{name}}].({severity}) output: {file_name}:{line_number} {message}, again: {message} ({function_name}()){'  # noqa
     name = 'test_logging_output_format_long'
-    action = launch_test.add_fixture_action(ld, ExecuteProcess(
+    launch_description.add_action(ExecuteProcess(
         cmd=[executable], env=env_long, name=name, output='screen'
     ))
-    output_file = os.path.join(os.path.dirname(__file__), name)
-    launch_test.add_output_test(
-        ld, action, create_output_test_from_file(output_file)
-    )
+    processes_to_test.append(name)
 
     env_edge_cases = dict(os.environ)
     # This custom output is to check different edge cases of the output format string parsing.
     env_edge_cases['RCUTILS_CONSOLE_OUTPUT_FORMAT'] = '{}}].({unknown_token}) {{{{'
     name = 'test_logging_output_format_edge_cases'
-    action = launch_test.add_fixture_action(ld, ExecuteProcess(
+    launch_description.add_action(ExecuteProcess(
         cmd=[executable], env=env_edge_cases, name=name, output='screen'
     ))
-    output_file = os.path.join(os.path.dirname(__file__), name)
-    launch_test.add_output_test(
-        ld, action, create_output_test_from_file(output_file)
-    )
+    processes_to_test.append(name)
 
     env_no_tokens = dict(os.environ)
     # This custom output is to check that there are no issues when no tokens are used.
     env_no_tokens['RCUTILS_CONSOLE_OUTPUT_FORMAT'] = 'no_tokens'
     name = 'test_logging_output_format_no_tokens'
-    action = launch_test.add_fixture_action(ld, ExecuteProcess(
+    launch_description.add_action(ExecuteProcess(
         cmd=[executable], env=env_no_tokens, name=name, output='screen'
     ))
-    output_file = os.path.join(os.path.dirname(__file__), name)
-    launch_test.add_output_test(
-        ld, action, create_output_test_from_file(output_file)
-    )
+    processes_to_test.append(name)
 
     env_time_tokens = dict(os.environ)
     # This custom output is to check that time stamps work correctly
     env_time_tokens['RCUTILS_CONSOLE_OUTPUT_FORMAT'] = "'{time}' '{time_as_nanoseconds}'"
     name = 'test_logging_output_timestamps'
-    action = launch_test.add_fixture_action(ld, ExecuteProcess(
+    launch_description.add_action(ExecuteProcess(
         cmd=[executable], env=env_time_tokens, name=name, output='screen'
     ))
-    output_file = os.path.join(os.path.dirname(__file__), name)
-    launch_test.add_output_test(
-        ld, action, create_output_test_from_file(output_file)
+    processes_to_test.append(name)
+
+    launch_description.add_action(
+        OpaqueFunction(function=lambda context: ready_fn())
     )
 
-    launch_service = LaunchService()
-    launch_service.include_launch_description(ld)
-    return_code = launch_test.run(launch_service)
-    assert return_code == 0, 'Launch failed with exit code %r' % (return_code,)
+    return launch_description, {'processes_to_test': processes_to_test}
+
```

Comment:
I suspect the lack of 'active' tests here will cause a issue, but it might be a problem that needs to be fixed in launch_testing.  I'm suspicious that in the absence of tests, the test thread can send the 'shutdown' signal to the LaunchService before it's finished launching the processes, which may cause the test runner to hang waiting for the LaunchService to terminate.  For robustness, it might be worth waiting for the process_to_test to terminate.